native/cocos/renderer/gfx-vulkan/VKDevice.cpp

```diff
@@ -503,27 +507,43 @@ void CCVKDevice::doDestroy() {
     }
     _depthStencilTextures.clear();
 
+    if (_gpuDevice) {
+        _gpuDevice->defaultBuffer = nullptr;
+        _gpuDevice->defaultTexture = nullptr;
+        _gpuDevice->defaultTextureView = nullptr;
+        _gpuDevice->defaultSampler = nullptr;
+    }
+
     CC_SAFE_DESTROY_AND_DELETE(_queryPool)
     CC_SAFE_DESTROY_AND_DELETE(_queue)
     CC_SAFE_DESTROY_AND_DELETE(_cmdBuff)
+
+    if (_gpuDevice) {
+        uint32_t backBufferCount = _gpuDevice->backBufferCount;
+        for (uint32_t i = 0; i < backBufferCount; i++) {
+            CC_SAFE_DELETE(_gpuStagingBufferPools[i])
+            CC_SAFE_DELETE(_gpuFencePools[i])
+        }
+    }
```

Comment:
Shall we use unique_ptr here? We might change them in future pr.

Reply:
Yes, it's better to use `unique_ptr`. I'm working on it.